features/step_definitions/your_contributions_steps.rb

```diff
@@ -2,22 +2,35 @@
   step 'I am on step 1 of the WPCC homepage'
 end
 
-Given(/^I complete the first step and move on to second step$/) do
+Given(/^I complete the your details form and move to your contributions page$/) do
   step 'I fill in the age, gender, salary and frequency fields'
   step 'I click on "My employer makes contributions on part of my salary"'
   step 'I click the Next button'
 end
 
-When(/^I click edit on the first step section$/) do
-  find_link('edit').click
+Given(/^I complete the your details and your contributions forms and move to your results page$/) do
+  step 'I fill in the age, gender, salary and frequency fields'
+  step 'I click on "My employer makes contributions on part of my salary"'
+  step 'I click the Next button'
+  save_and_open_page
+  click_button
+end
+
+When(/^I click edit on the your details summary$/) do
+  sections = page.find_all('.section--details')
+  within(sections.first) do
+    find_link('edit').click
```

Comment:
Could we possibly give the edit link a unique class name? This will save us the extraneous code for finding all elements within the `.section--details` class before proceeding to find the first element and then click on it. 
The test will start failing if any UI change puts in an element that precedes the `edit` link.